core/src/main/java/com/linecorp/armeria/client/limit/AbstractConcurrencyLimitingClient.java

```diff
@@ -84,30 +83,43 @@ protected AbstractConcurrencyLimitingClient(Client<I, O> delegate, int maxConcur
      */
     protected AbstractConcurrencyLimitingClient(Client<I, O> delegate,
                                                 int maxConcurrency, long timeout, TimeUnit unit) {
-        super(delegate);
-
-        validateAll(maxConcurrency, timeout, unit);
+        this(delegate, ConcurrencyLimit.builder()
+                .maxConcurrency(maxConcurrency)
+                .timeout(timeout, unit)
+                .build());
+    }
```

Comment:
Could you check the indentation? This should be:
```suggestion
        this(delegate, ConcurrencyLimit.builder()
                                       .maxConcurrency(maxConcurrency)
                                       .timeout(timeout, unit)
                                       .build());
    }
```